modules/nf-core/samtools/view/main.nf

```diff
@@ -34,6 +34,9 @@ process SAMTOOLS_VIEW {
                     args.contains("--output-fmt bam") ? "bam" :
                     args.contains("--output-fmt cram") ? "cram" :
                     input.getExtension()
+    def index_type = task.ext.index_type ?: ''
+    def output_name = args.contains("--write-index") ?  "${prefix}.${file_type}##idx##${prefix}.${file_type}.${index_type}" :
```

Comment:
This output name confuses me. It has prefix twice and file type twice and special characters. Is this really required?

Reply:
please see the comments from samtools devs, this is how they've implemented the functionality https://github.com/samtools/samtools/issues/1196

Reply:
Can we add a comment to this effect in the code please.